### DISCUSSION TITLE
Using WebView2 in Windows PE?

### DISCUSSION THREAD

**Discussion by mtniehaus** [MicrosoftEdge/WebView2Feedback#4293] [Q&A] [answered]
Windows PE (used to install Windows OSes) contains only the IE11 engine and MSHTA.EXE; we would like to use a "real" browser control and have been experimenting with WebView2 but not having much luck running it in Windows PE.  When starting to load, we get a COMException:

![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/16982296/e3215f1b-6b07-40a4-8953-92895c71ebcd)

Any idea what we're running into here?  This is an x64 OS with no wow64 support (so no ability to run 32-bit code).

The same Windows PE OS can use CefSharp (.NET 8 + WPF) without any issues.

**Comment by victorhuangwq** [comment, marked as answer]
WebView2 on Windows PE is not currently officially supported. That being said, could @lflores-ms take a look if you have time?

**Comment by icnocop** [comment]
Any updates or workarounds? Thank you.

**Comment by icnocop** [comment]
Related: https://github.com/MicrosoftEdge/WebView2Feedback/discussions/4947 and https://github.com/MicrosoftEdge/WebView2Feedback/issues/4951

**Reply by mtniehaus** [reply]
I would really like to get this to work, so let me throw in my progress:

* Windows PE runs as LocalSystem, which Edge does not like.  Add "--allow-run-as-system" to get around that,
* Windows PE is missing Media Foundation files (mf.dll, mfplat.dll), copy those from a Windows install into X:\windows\system32.
* Windows PE doesn't have a user profile folder.  Specify something like --user-data-dir="x:\temp" (needs to exist, create it first).

I've experimented with a variety of other options, but they all seem to fail.  "--headless" gets it to run, but without a UI, so it seems very likely that this is GPU-related.

I've attached a log from my most recent attempt.

[logfile.txt](https://github.com/user-attachments/files/21875800/logfile.txt)
